### PR TITLE
[docs] Use Grammarly to fix some typo

### PR DESCRIPTION
**Pull request by alexfauquette**
I tried a Grammarly extension to help catch English errors in the doc. @samuelsycamore could you check that I'm not introducing  new ones

### CODE REVIEW

**Comment by siriwatknp** [comment]
👋 The [migration PR](https://github.com/mui/mui-x/pull/3730) has been merged.

Please follow these steps to make sure that the contents are all updated. (Sorry for the inconvenience)

1. pull latest master from upstream to your branch
2. if your PR has changes on the `*.md` or demo files, you are likely to encounter conflict because all of them have been moved to the new folder.
   2.1 revert the change on those markdown files you did
   2.2 pull latest master from upstream (you should not see conflict)
   2.3 make the changes again at `docs/data/material/*`
3. run `yarn docs:api`
    - you might see the changes in `docs/pages/material/api/*` if your PR touches some of the components
    - it is okay if there is no changes

If you are struggle with the steps above, feel free to tag @siriwatknp

**Comment by samuelsycamore** [comment]
Looks good overall. I see lots of improvements here. A few suggestions _(in italics)_:

128. _This enables you to customize_ how the entered value is stored in the row.

160. Filtering and sorting _do_ not rely on the formatted value.

444. Pinned (or frozen, locked, or sticky) columns are columns that are visible _at all times_ while the user scrolls the grid horizontally.

478. While the `initialState` prop only works for setting pinned columns during the initialization, the `pinnedColumns` prop allows _you to modify which columns are pinned at any time_.

527. _You can modify this default behavior with column spanning._

270. _The example below demonstrates how to achieve server-side filtering._

70. You can override the **headerName** or any property of the `GridColDef` interface, except the `field`, the `type`, and the properties related to inline **edition.** I'm not familiar with the context here, but should this be **addition**?

96. - Add _an_ `onPageChange` callback to load the rows when the page changes

File: docs/data/data-grid/columns/columns.md [hunk 1]
```diff
@@ -524,7 +524,7 @@ Grouping columns allows you to have multiple levels of columns in your header an
 > 👍 Upvote [issue #192](https://github.com/mui/mui-x/issues/192) if you want to see it land faster.
 
 Each cell takes up the width of one column.
-Column spanning allows to change this default behavior.
+Column spanning allows modifying this default behavior.
 It allows cells to span multiple columns.
```
Comment (alexfauquette) [inline docs/data/data-grid/columns/columns.md:528]:
```suggestion
It allows cells to span multiple rows.
```

**Comment by alexfauquette** [comment]
@samuelsycamore Thanks for the feedback. For making suggestions, github has an amazing tool

In "Files changed", on the left of each line of code, you have a button to make a comment. And in the comment editor, you have this button
image
It adds the line of code you clicked. You then modified it, and other will see your suggestion as a git diff in its context


Peek 2022-02-10 09-31

**Comment by alexfauquette** [comment]
About your point 134. the context is the following:

Users are able to group rows, and this part of the doc is about how to customize the columns that allow to extend or collapse a group of rows. Values of this column can not be edited by the end-user, so we say to developers that they can not override "properties related to inline edition"

**Comment by samuelsycamore** [comment]
Thank you so much for that walkthrough @alexfauquette! I'll be sure to use that tool going forward. 👍

**Pull request merged**